### DISCUSSION TITLE
Hard-Coded Opener

### DISCUSSION THREAD

**Discussion by shahern004** [Hekili/hekili#655] [Ideas]
First of all, huge fan of the addon. Long time user. I was wondering, is it possible to edit the APL in the addon to hard-code in an opener rotation, and then the addon falls into its priority list?

**Comment by Hekili** [comment]
Nope.  There are problems with a hardcoded opener (a strict sequence, in SimulationCraft terms) that make this not viable.

In game, I really have no way of knowing that you pushed a button *because* the addon recommended it.

I would have no way of knowing if you wanted to actually continue the sequence if you got stunned or pressed something else (let's say you hit a defensive or self-heal... does that mean you're done with the sequence, that we should drop out, that you should restart?).

For most openers, one could use priority logic to ensure an opening sequence is successfully completed, but it can get complex depending on whether talents/runeforges/conduits/etc. impact that opener.
